core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1210,8 +1219,14 @@ public ScheduledFuture<?> scheduleInEventLoop(Runnable runnable, long duration,
         }, duration, timeUnit);
     }
 
+    public void updateTickingState(float tickRate, boolean frozen) {
+        tickThread.cancel(true);
+        this.tickingFrozen = frozen;
+        tickThread = eventLoop.scheduleAtFixedRate(this::tick, Math.round(1000 / tickRate), Math.round(1000 / tickRate), TimeUnit.MILLISECONDS);
+    }
```

Comment:
1. we wouldn't want to interrupt a tick, and it shouldn't happen anyway because TickingStatePacket is handled on the session's event loop. i would do `tickThread.cancel(false)` just to be clear.
2. it would be nice to apply a tickrate minimum of 1, like the vanilla client does

Reply:
do you think we should also implement a maximum just to be safe? it would also be the same as vanilla because afaik theres a cap there too

Reply:
Fixed in latest commit